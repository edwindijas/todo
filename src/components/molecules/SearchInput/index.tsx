import SearchIcon from '@/components/atoms/Icon/Search'
import TextInput, { type InputProps } from '@/components/atoms/TextInput'

const SearchInput = ({
  label = 'Search',
  intent = 'secondary',
  ...props}: SearchInputProps):
  JSX.Element =>
<TextInput label={label} {...props} intent={intent} startIcon={<SearchIcon />} />

interface SearchInputProps extends Omit<InputProps, 'label'> {
  label?: string
}

export default SearchInput