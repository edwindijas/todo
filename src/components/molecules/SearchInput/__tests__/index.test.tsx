import render from "@/utils/test-utils/customRender";
import SearchInput from "..";

it('renders test input correctly', () => {
  const {container} = render(<><SearchInput label='Hello World' /><SearchInput /></>)
  expect(container).toMatchSnapshot()
})