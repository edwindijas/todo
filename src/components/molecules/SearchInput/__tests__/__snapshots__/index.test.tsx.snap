// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders test input correctly 1`] = `
<div>
  <label
    class="undefined  flex relative cursor-text border-[#A6B7D0] bg-[#F4F8FC]  fill-blue-desaturated  border rounded-lg h-12 px-2"
  >
    <div
      class="shrink-0"
    >
      <svg
        class=" w-full h-full"
        data-testid="component-atom-icon"
      />
    </div>
    <div
      class="relative"
    >
      <span
        class="absolute flex top-0 bottom-0 left-0 items-center text-blue-desaturated opacity-100"
      >
        Hello World
      </span>
      <input
        class="block outline-none w-full h-full bg-transparent h-12"
        value=""
      />
    </div>
  </label>
  <label
    class="undefined  flex relative cursor-text border-[#A6B7D0] bg-[#F4F8FC]  fill-blue-desaturated  border rounded-lg h-12 px-2"
  >
    <div
      class="shrink-0"
    >
      <svg
        class=" w-full h-full"
        data-testid="component-atom-icon"
      />
    </div>
    <div
      class="relative"
    >
      <span
        class="absolute flex top-0 bottom-0 left-0 items-center text-blue-desaturated opacity-100"
      >
        Search
      </span>
      <input
        class="block outline-none w-full h-full bg-transparent h-12"
        value=""
      />
    </div>
  </label>
</div>
`;
